docs/design/02-topologies.md

```diff
@@ -0,0 +1,288 @@
+
+# Overview
+
+- Will have new `metal configure` command group for configuring values to be
+  used within config files for all nodes, a group of nodes, or an individual
+  node:
+  - `metal configure all`
+  - `metal configure group GROUP`
+  - `metal configure node NODE`
+
+- These commands may also perform other behaviour specific to the type of
+  resource being configured.
+
+- Note that these commands will be just one interface for configuring the
+  'topology' of a cluster; it should be possible for alternative interfaces,
+  such as a future web interface or manual editing, to modify the same files,
+  and the loading of these files for use when templating will be independent of
+  the `configure` commands.
+
+- There will be a `configure.yaml` file in the repo root, specifying
+  configuration questions to be used within these commands or alternative
+  interfaces.
+
+- The `configure` commands will save the entered answers to a file in
+  `/var/lib/metalware/config`, specific to the resource being configured.
+
+- Repo configs will be able to use a new object in the magic `alces` namespace,
+  `alces.config.$question_identifier`, where `$question_identifier` is the
```

Comment:
@ste78 - yep that may be the best choice, I'll add to my last commit to mention that in the doc